form-clone.js

```diff
@@ -68,14 +68,17 @@ $.fn.formClone = function(options){
 					$(this).insertAfter($div);
 
 					// Pass in "all" to copy events
-					$cloned = $div.clone($(this).data('all')).insertAfter($(this));
+					$cloned = $div.clone($(this).data('all'));
 
 					$formCloneRemoveButton(el, $(this).data('buttonCss')).insertBefore($(this));
 
 					// Set data.all-values to copy values
 					if (!$(this).data('all-values')){
-						$cloned.inputBlankArray().find('input:not([type="submit"])').val('');
+						$cloned.inputBlankArray().find('input:not([type="submit"]):not([type="radio"]):not([type="checkbox"])').val('');
+						$cloned.find('input[type="radio"], input[type="checkbox"]').prop('checked', false);
 					}
+					// do insert after un-check any radios so we don't insert checked radios and so possibly unset another already in the page

```

Comment:
The select here should only change items in the `$cloned` data, can you explain what you observed with items outside of this scope being altered?

Reply:
The cloned radio button had the same name as the one it was cloned from (clonee). Thus, if the clonee is checked, when the cloned one is inserted we have two radios with the same name, and thus the same 'group' checked, so the browser made it so only the new radio was checked (and the code then unchecked it).